.github/workflows/build-keystone-operator.yaml

```diff
@@ -11,177 +11,10 @@ env:
-      run: |
-        echo "latesttag=${{ steps.branch-name.outputs.current_branch }}-latest" >> $GITHUB_ENV
-
-    - name: Build keystone-operator-bundle using buildah
-      id: build-keystone-operator-bundle
-      uses: redhat-actions/buildah-build@v2
-      with:
-        image: keystone-operator-bundle
-        tags: ${{ env.latesttag }} ${{ github.sha }}
-        containerfiles: |
-          ./bundle.Dockerfile
-
-    - name: Push keystone-operator To ${{ env.imageregistry }}
-      uses: redhat-actions/push-to-registry@v2
-      with:
-        image: ${{ steps.build-keystone-operator-bundle.outputs.image }}
-        tags: ${{ steps.build-keystone-operator-bundle.outputs.tags }}
-        registry:  ${{ env.imageregistry }}/${{ env.imagenamespace }}
-        username: ${{ secrets.QUAY_USERNAME }}
-        password: ${{ secrets.QUAY_PASSWORD }}
-
-  build-keystone-operator-index:
-    needs: [ check-secrets, build-keystone-operator-bundle ]
-    name: keystone-operator-index
-    runs-on: ubuntu-latest
-    if: needs.check-secrets.outputs.have-secrets == 'true'
-
-    steps:
-    - name: Checkout keystone-operator repository
-      uses: actions/checkout@v2
-
-    - name: Get branch name
-      id: branch-name
-      uses: tj-actions/branch-names@v5
-
-    - name: Set latest tag for non main branch
-      if: "${{ steps.branch-name.outputs.current_branch != 'main' }}"
-      run: |
-        echo "latesttag=${{ steps.branch-name.outputs.current_branch }}-latest" >> $GITHUB_ENV
-
-    - name: Install opm
-      uses: redhat-actions/openshift-tools-installer@v1
-      with:
-        source: github
-        opm: 'latest'
-
-    - name: Log in to Red Hat Registry
-      uses: redhat-actions/podman-login@v1
-      with:
-        registry: ${{ env.imageregistry }}
-        username: ${{ secrets.QUAY_USERNAME }}
-        password: ${{ secrets.QUAY_PASSWORD }}
-
-    - name: Create index image
-      run: |
-        pushd "${GITHUB_WORKSPACE}"/.github/
-        chmod +x "create_opm_index.sh"
-        "./create_opm_index.sh"
-        popd
-      env:
-        REGISTRY:  ${{ env.imageregistry }}/${{ env.imagenamespace }}
-        GITHUB_SHA: ${{ github.sha }}
-        BUNDLE_IMAGE: keystone-operator-bundle
-        INDEX_IMAGE_TAG: ${{ env.latesttag }}
-        INDEX_IMAGE: keystone-operator-index
-
-    - name: Push keystone-operator-index To ${{ env.imageregistry }}
-      uses: redhat-actions/push-to-registry@v2
-      with:
-        image: keystone-operator-index
-        tags: ${{ env.latesttag }} ${{ github.sha }}
-        registry:  ${{ env.imageregistry }}/${{ env.imagenamespace }}
-        username: ${{ secrets.QUAY_USERNAME }}
-        password: ${{ secrets.QUAY_PASSWORD }}
+  call-build-workflow:
+    uses: openstack-k8s-operators/openstack-k8s-operators-ci/.github/workflows/reusable-build-operator.yaml@main
+    with:
+      operator_name: keystone
+      go_version: 1.19.x
+      operator_sdk_version: 1.31.0
+    secrets: inherit
```

Comment:
Hit an issue here where secrets aren't inherited across an org. So if the workflow above is in 'openstack-k8s-operators' it doesn't allow my personal 'dprince' secrets to be passed into that workflow unless I modify the uses: line above to be from my own repo (this is how I had it for testing and only just noticed it) @viroel @stuggi @gibizer Looking into options to support this for "forked" workflows. Landing this now wouldn't break our upstream builds, but it would definately impact those of us use have secrets set in our forked repositories

Reply:
Okay, Figured it out. Just need to be explicit and not use 'inherit'. So this works`    secrets:
      IMAGENAMESPACE: ${{ secrets.IMAGENAMESPACE }}
      QUAY_USERNAME: ${{ secrets.QUAY_USERNAME }}
      QUAY_PASSWORD: ${{ secrets.QUAY_PASSWORD }}
      REDHATIO_USERNAME: ${{ secrets.REDHATIO_USERNAME }}
      REDHATIO_PASSWORD: ${{ secrets.REDHATIO_PASSWORD }}
`